### PR TITLE
QNTM-3872

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose

Task : https://jira.autodesk.com/browse/QNTM-3872

This PR fixes the issue with custom nodes issue where the InputSymbol is not deserialized correctly for unknown types.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @QilongTang

### CODE REVIEW

File: src/DynamoCore/Graph/Workspaces/SerializationConverters.cs [hunk 1]
```diff
@@ -399,6 +399,9 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             var connectors = obj["Connectors"].ToObject<IEnumerable<ConnectorModel>>(serializer);
 
             var info = new WorkspaceInfo(guid.ToString(), name, description, Dynamo.Models.RunType.Automatic);
+            //https://jira.autodesk.com/browse/QNTM-3872 
+            //Category should be set explictly from custom node workspace.
+            info.Category = obj["Category"].Value<string>();
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: why is this necessary? - shouldn't the node workspaceInfo contain this?

File: test/DynamoCoreTests/CustomNodes.cs [hunk 2]
```diff
@@ -1089,5 +1090,24 @@ public void Regress3786_InputNodesDeserilization()
             Assert.IsNotNull(node);
             Assert.AreEqual("var1: Autodesk.DesignScript.Geometry.Point", node.Parameter.ToNameString());
         }
+
+        [Test]
+        public void Regress3872_InputNodesDeserilization()
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/CustomNodes.cs]: 👍

**Comment by mjkkirschner** [comment]
one small question and one spelling comment.

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs [hunk 3]
```diff
@@ -433,15 +433,20 @@ public string InputSymbol
                                 Properties.Resources.WarningCannotFindType,
                                 identifierNode.datatype.Name);
                             this.Warning(warningMessage);
+                            //https://jira.autodesk.com/browse/QNTM-3872
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: unknown (spelling)

File: src/DynamoCore/Graph/Workspaces/SerializationConverters.cs (hunk 1 above)
Comment (ramramps) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: nope..Category is not in the constructor..in XML, this is done explicitly. `FromJSONDocument` does it correctly. again, in `serializtionConverters` this is set to different value.

Comment (QilongTang) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: Same question, I think the cate is included in workspace info, you can see how customNodeWorkspace got initialized at https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Workspaces/CustomNodeWorkspaceModel.cs#L90

**Comment by ramramps** [comment]
Running self-serve CI

File: src/DynamoCore/Graph/Workspaces/SerializationConverters.cs (hunk 1 above)
Comment (ramramps) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: This is after deserialization.

Comment (mjkkirschner) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: hmm, but you can see above that workspaceInfo is not passed a category... I think Ram is right... wow.

Comment (QilongTang) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: Wait how could **cate** be deserialized correctly before this change?

Comment (mjkkirschner) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: yeah, I am wondering the same thing... Ram is saying it was only working with xml custom nodes but I am not sure about that...

Comment (ramramps) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: It is not actually. `Open the xml dyf - save - open - save : category will be null`

Comment (mjkkirschner) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: if thats true... it is bad.

Comment (ramramps) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: you can test on the master branch..but I did the testing, and ran into this.

Comment (ramramps) [reply src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: I think Andreas ran into the same issue....

**Comment by ramramps** [comment]
Self serv CI passed. merging this
